api/src/main/java/org/openmrs/module/kenyaemr/reporting/library/MOH706/Moh706LabCohortLibrary.java

```diff
@@ -35,4 +38,182 @@ public CohortDefinition getAllUrineAnalysisGlucoseTestsPositives() {
         );
         return sql;
     }
+
+    public CohortDefinition getAllUrineAnalysisKetonesTestsPositives() {
+        SqlCohortDefinition sql = new SqlCohortDefinition();
+        sql.setName("Get urine analysis patients - ketones");
+        sql.addParameter(new Parameter("startDate", "Start Date", Date.class));
+        sql.addParameter(new Parameter("endDate", "E    nd Date", Date.class));
+        sql.setQuery("SELECT d.patient_id FROM kenyaemr_etl.etl_patient_demographics d\n" +
+                "                             INNER JOIN kenyaemr_etl.etl_laboratory_extract x ON x.patient_id = d.patient_id\n" +
+                "                             WHERE x.lab_test = 1305 AND x.test_result = 1302 AND date(x.visit_date) BETWEEN :startDate AND :endDate\n" +
```

Comment:
The concept ID 856 and 1305 are for viral load. Please substitute with the correct ones for urinalysis